### ISSUE TITLE
Tainted samson dies if he takes a devil deal item that gives health back and making it having no health left while at the end of Berserk mode

### ISSUE THREAD

**Issue by Rchardon** [UnderscoreKilburn/repentance-issues#1962]
And it should give the health back like on every other characters and not make him die

**Comment by Zamiell** [comment]
This is because the collectible is queued and the check for death happens before collectibles are dequeued.
The order needs to be switched.

**Comment by ZMBanshee** [comment]
I don't really see this as much different from all the other cases where you can die during the item grab animation. It does make you wonder if any of those cases should be a thing either, though.

**Comment by Zamiell** [comment]
Yeah. Ideally, dying to a fire after taking e.g. Dead Cat would not result in death.

**Comment by Rchardon** [comment]
> I don't really see this as much different from all the other cases where you can die during the item grab animation. It does make you wonder if any of those cases should be a thing either, though.

This case is special because you don't get hit while you take the item. I can understand dying to a fire because you took a hit but there you just end a gameplay mode
